docs/website/docs/user-guides/v3-migration-guide.md

```diff
@@ -0,0 +1,88 @@
+---
+title: Migrate from v2 to v3
+sidebar_position: 2
+---
+
+### Where are `odo push` and `odo watch` commands?
+`odo push` and `odo watch` have been replaced in v3 by a single command - `odo dev`. It does the job of two commands 
+in a single command, and also allows the user to not watch if `--no-watch` flag is passed. 
+
+In v2, if you wanted to automatically sync the code on developer system with the application running on a Kubernetes
+cluster, you had to perform two steps - first `odo push` to start the application on the cluster, and second `odo watch`
+to automatically sync the code. In v3, `odo dev` performs both actions with a single command.
+
+`odo dev` is not _just_ a replacement for these two commands. It's also different in behaviour in that, it's a 
+long-running process that's going to block the terminal. Hitting `Ctrl+c` will stop the process and cleanup the 
+component from the cluster. In v2, you had to use `odo delete`/`odo component delete` to delete inner loop resources 
+of the component from the cluster.
+
+### Migrate existing odo component from v2 to v3
+If you have created an odo component using odo v2, this section will help you move it to use odo v3.
```

Comment:
It could be helpful to mention that users can run `odo list` in v3, to see which components are running and in which version.

Reply:
Makes sense. But I have added this as fourth point in this section instead of the brief intro because it is something that is available to the user after they have downloaded odo v3.